packages/shadow-dom/src/findInsertionPoint.ts

```diff
@@ -39,3 +39,32 @@ export function findInsertionPoint(
 
   return null;
 }
+
+export function findShadowRootInsertionPoint(
+  renderer: GriffelShadowDOMRenderer,
+  shadowRoot: ShadowRoot,
+  rendererTargetStyleSheet: ExtendedCSSStyleSheet | null,
+  insertionPoint?: CSSStyleSheet,
+): CSSStyleSheet | null {
+  const styleSheets = shadowRoot.adoptedStyleSheets;
+
+  if (renderer.adoptedStyleSheets.length < 1) {
+    return insertionPoint ? styleSheets[styleSheets.indexOf(insertionPoint) + 1] : null;
+  }
```

Comment:
I tried to understand when it's needed, but failed... Can you please a UT for this scenario? As so far it looks redundant.

Reply:
You're right, it is redundant!

When I was working through this I had it as an "early out" conditional but this case is handled by the `while` loop just as well.